src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -135,7 +143,7 @@ BlockIO InterpreterDropQuery::executeToTable(
             DatabaseCatalog::instance().waitTableFinallyDropped(table_id.uuid);
         else if (query.kind == ASTDropQuery::Kind::Detach)
         {
-            if (auto * atomic = typeid_cast<DatabaseAtomic *>(database.get()))
+            if (auto * atomic = dynamic_cast<DatabaseAtomic *>(database.get()))
```

Comment:
In other places it's a check for UUID to detect Atomic DB. Why dynamic_cast here?